### ISSUE TITLE
Minor Upgrade Fails upon docker image update

### ISSUE THREAD

**Issue by vs102000** [bitnami/bitnami-docker-owncloud#63]
when pulling the newest 10.4.1 image, the nami bootstrap procedure checks occ status and compares first_install_version with the current version. Within the the is a diff in the minor version, it switches to maintenance mode and tries occ upgrade in any case. This fails since occ upgrade throws an error when noting is to upgrade and the system is in maintenance mode.

Tricky part: this never happend in a fresh install, since first_install_version is always equal to the current version.

To reproduce: start from e.g. docker image 10.4.1-debian-10-r0 and then switch to 10.4.1-debian-10-r73 with a persistent volume and db.

**Comment by beltran-rubo** [comment]
Hi,

Thanks for reporting it. I tried to reproduce it with the same tags but I could not reproduce it. We have this logic to detect if it requires to run the "occ upgrade" process or not:

```
 const firstInstallVersion = semver.coerce(status.first_install_version);
  const currentVersion = semver.coerce(status.version);
  if (semver.gt(currentVersion, firstInstallVersion)) {
       [[occ upgrade actions]]
```

That means if the Owncloud version is the same (10.4.1) it should not go through that upgrade process.  Could you let us know if you upgrade from a previous ownCloud version?

**Comment by vs102000** [comment]
Hi,

the problem is that the firstInstallVersion value is stored in the owncloud DB. So for example if you start with a docker image which references  10.4.1.1 and update to a new docker image using e.g.  10.4.1.3 then the occ status is reporting:

  - installed: true
  - first_install_version: 10.4.1.1
  - version: 10.4.1.3
  - versionstring: 10.4.1
  - edition: Community

When you do occ upgrade, owncloud is upgraded, but occ status will continue to report first_install_version: 10.4.1.1

In other words, I think the logic for you upgrade check is wrong and should not rely on first_install_version.

**Comment by marcosbc** [comment]
Thanks for pointing that out. It looks like indeed, the logic is wrong and should be changed to compare the known version (contained in the metadata for the container initialization scripts), with the "version" field.

I've created an internal task to work on this issue. Unfortunately I cannot give an ETA for when this will be fixed, but we'll keep you posted if there's any update.

**Comment by marcosbc** [comment]
Hi @vs102000, just wanted to let you know that we have just released [ownCloud `10.6.0-debian-10-r18`](https://github.com/bitnami/bitnami-docker-owncloud#1060-debian-10-r18) which includes a fix for this issue. In addition, the initialization logic has been refactored and the source code is now available in the [`rootfs` folder](https://github.com/bitnami/bitnami-docker-owncloud/tree/master/10/debian-10/rootfs) in this repo, plus lots of other changes.
